components/api_request_helper/api_request_helper.cc

```diff
@@ -355,25 +326,54 @@ void APIRequestHelper::URLLoaderHandler::OnDataReceived(
 
 void APIRequestHelper::URLLoaderHandler::OnComplete(bool success) {
   DCHECK(data_completed_callback_);
-  DVLOG(1) << "[[" << __func__ << "]]"
-           << " Response completed\n";
+  VLOG(1) << "[[" << __func__ << "]]"
+          << " Response completed\n";
 
   std::move(data_completed_callback_)
-      .Run(api_request_helper_->ToAPIRequestResult(std::move(url_loader_)),
-           success);
-
-  api_request_helper_->Cancel(this);
+      .Run(ToAPIRequestResult(std::move(url_loader_)), success);
+  // If we're still parsing, we don't want to Cancel as that will delete
+  // the decoder instance for this request. Wait until all decoder operations
+  // are complete.
+  DCHECK_LT(-1, current_decoding_operation_count_);
```

Comment:
```suggestion
  DCHECK_LE(0, current_decoding_operation_count_);
```

Reply:
This has changed now and moved to `TryDelete`